index.ts

```diff
@@ -16,7 +16,49 @@ interface MemoryHandle {
   readString(): string;
 }
 
-type Input = string | ArrayBuffer | object;
+// Provides access to data in Extism memory
+export class MemoryView {
```

Comment:
If [`DataView`](https://mdn.io/dataview) is available in quickjs, subclassing it can be a good way to provide a bunch of ancillary buffer data access functions – the only tricky thing is that, if the backing arraybuffer is `null` we'd have to use a sentinel "empty" array buffer.

Something like:

```ts
const NULL = new ArrayBuffer(0)

export class MemoryView extends DataView {
  static #decoder = new TextDecoder();
  #isNull: bool
  constructor(memory?: MemoryHandle) {
    super(memory ? memory.readBytes() : NULL)
    this.#isNull = !memory
  }

  isEmpty() {
    return this.buffer.byteLength === 0
  }

  text(): string {
    return MemoryView.#decoder.decode(this.buffer)
  }

  // ...
}
```

(OTOH it means we'd _always_ load up the backing memoryhandle as an ArrayBuffer on initialization. I don't have a great read on whether this is an okay cost to pay– how often would users receive a memory view without consulting its contents? Also, I'm not sure if quickjs supports `DataView` – if it doesn't, ignore this comment, haha 😅)

Reply:
Thanks, I will look into this!  I was initially thinking about this type as a way to lazily convert the memory to JS values, but I don't think doing the conversion to ArrayBuffer up-front is really an issue, and since it's possible to return encoded integer values, having those extra methods available could be pretty helpful.

Reply:
It looks like quickjs does support `DataView` so I updated `MemoryView` to extend `DataView` in https://github.com/dylibso/xtp-test-js/pull/12/commits/831eb9423c22d224fd4a51a617a5b29c0f787964

I don't know if we need to make the distinction between a `null` and empty `MemoryView` so I didn't include the `#isNull` field - but let me know if there's a use-case I'm not considering.

Reply:
That looks great! There might be _some_ distinction between `isNull` vs `isEmpty` –whether or not a handle was passed at all- but I'm in the same boat; I'm not sure if folks will care to distinguish those cases.